### PR TITLE
Redundancies in JvmName

### PR DESCRIPTION
**Pull request by JonathanStarup**
* Undid wrong autoformat i made previously
* Unused names in JvmName deleted
* Removed a dublicate method that calculated the ref name
* removed cases in getJvmType that resulted in classes that doesn't exist

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/jvm/AsmOps.scala [hunk 1]
```diff
@@ -541,7 +541,7 @@ object AsmOps {
     mv.visitInsn(ACONST_NULL)
 
     // Construct the proxy object.
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/AsmOps.scala:543]: Is this method even used?

**Comment by magnus-madsen** [comment]
Looks good.

File: main/src/ca/uwaterloo/flix/language/phase/jvm/AsmOps.scala (hunk 1 above)
Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/AsmOps.scala:543]: Its used in relation to function application but right now it uses `flix/runtime/ProxyObject` while the jvm name refers to `JvmName(List("dev", "flix", "runtime"), "ProxyObject")`. It seems like it isn't used somehow but it's definitely called.. Im not sure

Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/AsmOps.scala:543]: I've fixed the reference now. I'll look further into it when it gets its turn during the dsl iteration

**Pull request merged**